arangod/Aql/RestAqlHandler.cpp

```diff
@@ -67,6 +68,12 @@ RestAqlHandler::RestAqlHandler(ArangodServer& server, GeneralRequest* request,
   TRI_ASSERT(_queryRegistry != nullptr);
 }
 
+RestAqlHandler::~RestAqlHandler() {
+  if (_logContextQueryIdEntry) {
+    LogContext::Current::popEntry(_logContextQueryIdEntry);
```

Comment:
I see this `popEntry` here and the one in `shutdownExecute`. How to we prevent that both are executed when the query is finished? As far as I see the member `_logContextQueryIdEntry` is not set to `nullptr`...

Reply:
`popEntry` does a check first, and also sets the entry to a nullptr. Therefore it is safe to call it multiple times on the same item.
Here is its implementation:
```
inline void LogContext::Current::popEntry(EntryPtr& entry) noexcept {
  if (entry._entry != nullptr) {
    auto& local = LogContext::controlBlock();
    TRI_ASSERT(entry._entry == local._logContext._tail);
    local._logContext.popTail(local._entryCache);
    entry._entry = nullptr;
  }
}
```

Reply:
Understood. Thanks!